prov/rxm/src/rxm_ep.c

```diff
@@ -512,13 +516,21 @@ static ssize_t rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov
 		pkt_size = sizeof(*pkt) + pkt->hdr.size;
 	}
 
-	ret = fi_send(rxm_conn->msg_ep, pkt, pkt_size, desc_tx_buf, 0, tx_entry);
-	if (ret) {
-		FI_WARN(&rxm_prov, FI_LOG_EP_DATA, "fi_send for MSG provider failed\n");
-		goto err;
+	if (flags & FI_INJECT & ~FI_COMPLETION) {
+		ret = fi_inject(rxm_conn->msg_ep, pkt, pkt_size, 0);
```

Comment:
Sorry, rxm *does* support selective completion. :) The FI_COMPLETION flag is being handled when we report completion in rxm_finish_send().
>My suggestion is to use fi_sendmsg instead of fi_send and fi_inject. I meant that provider on which RXM based will decide how to data should sent.

If we have a good reason to let the underlying provider deal with the flags, we can do it. In current design, the flags are supposed to be handled in rxm itself.